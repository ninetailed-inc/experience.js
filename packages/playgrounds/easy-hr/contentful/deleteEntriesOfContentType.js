const contentfulManagement = require('contentful-management');

require('dotenv').config({
  path: `${process.env.PATH_TO_ENV_FILE}`,
});

const contentfulClient = contentfulManagement.createClient({
  accessToken: process.env.CONTENTFUL_MANAGEMENT_TOKEN,
});

async function unpublishAndDeleteEntry(entry) {
  try {
    if (entry.isPublished()) {
      console.info(`Deleting published entry with id: ${entry.sys.id}`);
      await entry.unpublish().then((e) => {
        return e.delete();
      });
    } else {
      console.info(`Deleting unpublished entry with id: ${entry.sys.id}`);
      await entry.delete();
    }
  } catch (e) {
    console.log(e);
  } finally {
    console.info(`Deleted entry: ${entry.sys.id}`);
  }
}

contentfulClient
  .getSpace(process.env.NEXT_PUBLIC_CONTENTFUL_SPACE_ID)
  .then((space) => {
    space
      .getEnvironment(process.env.NEXT_PUBLIC_CONTENTFUL_ENVIRONMENT)
      .then((environment) => {
        console.info(`Accessing Environment: ${environment.sys.id}`);
        console.info(
          `Deleting Entries Of ContentType: ${process.env.CONTENT_TYPE} `
        );
        environment
          .getEntries({
            content_type: `${process.env.CONTENT_TYPE}`,
            limit: `${process.env.LIMIT}`,
          })
          .then(async (response) => {
            await Promise.all(response.items.map(unpublishAndDeleteEntry));
          });
      });
  });
