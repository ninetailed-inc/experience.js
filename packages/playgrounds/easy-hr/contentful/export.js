const contentfulExport = require('contentful-export');
const dotEnv = require('dotenv');
// https://github.com/contentful/contentful-export

dotEnv.config({ path: `${process.env.PATH_TO_ENV_FILE}` });
/**
 * Provide the deliveryToken if you want to export
 * just the latest published versions of your content
 * */
const exportOptions = {
  spaceId: process.env.CONTENTFUL_SPACE_ID,
  environmentId: process.env.CONTENTFUL_ENVIRONMENT,
  managementToken: process.env.CONTENTFUL_MANAGEMENT_TOKEN,
  /* deliveryToken: process.env.CONTENTFUL_TOKEN, */
  contentFile: process.env.CONTENTFUL_SPACE_DATA_LOCATION,
  includeDrafts: true,
};

if (
  !process.env.CONTENTFUL_SPACE_ID ||
  !process.env.CONTENTFUL_MANAGEMENT_TOKEN ||
  !process.env.CONTENTFUL_SPACE_DATA_LOCATION
) {
  throw new Error(
    [
      'Parameters missing...',
      'Please insert the following credentials into your .env.local.local file:',
      '- CONTENTFUL_SPACE_ID=XXX',
      '- CONTENTFUL_MANAGEMENT_TOKEN=CFPAT-XXX',
      '- CONTENTFUL_SPACE_DATA_LOCATION="PATH TO STORAGE DIRECTORY"',
      'Afterwards run the export command as follows:',
      '"npm run export" or "yarn export"',
    ].join('\n')
  );
}

contentfulExport(exportOptions)
  .then((result) => {
    console.log('Your space data:', result);
  })
  .catch((err) => {
    console.log('Oh no! Some errors occurred!', err);
  });
