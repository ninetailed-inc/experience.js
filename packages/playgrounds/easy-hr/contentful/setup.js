const contentfulImport = require('contentful-import');
const dotEnv = require('dotenv');
// https://github.com/contentful/contentful-import

dotEnv.config({ path: `${process.env.PATH_TO_ENV_FILE}` });

const importOptions = {
  spaceId: process.env.NEXT_PUBLIC_CONTENTFUL_SPACE_ID,
  environmentId: process.env.NEXT_PUBLIC_CONTENTFUL_ENVIRONMENT || 'master',
  managementToken: process.env.CONTENTFUL_MANAGEMENT_TOKEN,
  contentFile:
    process.env.CONTENTFUL_SPACE_DATA_LOCATION ||
    './packages/playgrounds/easy-hr/contentful/data/contentful-space-data.json',
};

if (
  !process.env.NEXT_PUBLIC_CONTENTFUL_SPACE_ID ||
  !process.env.CONTENTFUL_MANAGEMENT_TOKEN
) {
  throw new Error(
    [
      'Parameters missing...',
      'Please insert the following credentials into your .env.local file:',
      '- NEXT_PUBLIC_CONTENTFUL_SPACE_ID=XXX',
      '- CONTENTFUL_MANAGEMENT_TOKEN=CFPAT-XXX',
      'Afterwards run the setup command as follows:',
      '"npm run setup" or "yarn setup"',
    ].join('\n')
  );
}
contentfulImport(importOptions)
  .then(() => {
    return console.log('The content model of your space is set up!');
  })
  .catch((e) => {
    return console.error(e);
  });
