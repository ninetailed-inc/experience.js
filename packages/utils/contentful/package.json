{
  "name": "@ninetailed/experience.js-utils-contentful",
  "version": "7.5.1",
  "description": "Contentful utilities for Experience.js",
  "devDependencies": {
    "contentful": "^9.1.32",
    "@contentful/rich-text-types": "16.2.0"
  },
  "license": "MIT"
}
