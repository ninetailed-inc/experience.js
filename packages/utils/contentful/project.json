{
  "name": "utils-contentful",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/utils/contentful/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/utils/contentful",
        "tsConfig": "packages/utils/contentful/tsconfig.lib.json",
        "project": "packages/utils/contentful/package.json",
        "entryFile": "packages/utils/contentful/src/index.ts",
        "external": [],
        "rollupConfig": "@nrwl/react/plugins/bundle-rollup",
        "buildableProjectDepsInPackageJsonType": "dependencies",
        "format": ["cjs", "esm"],
        "assets": [
          {
            "glob": "packages/utils/contentful/README.md",
            "input": ".",
            "output": "."
          }
        ]
      },
      "dependsOn": [
        {
          "target": "build",
          "projects": "dependencies"
        }
      ]
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/utils/contentful/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/packages/utils/contentful"],
      "options": {
        "jestConfig": "packages/utils/contentful/jest.config.js",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
