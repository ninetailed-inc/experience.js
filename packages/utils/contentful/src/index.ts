export { AudienceMapper } from './lib/AudienceMapper';
export { ExperienceMapper } from './lib/ExperienceMapper';
export { isEntry } from './lib/isEntry';

export type {
  AudienceEntry,
  AudienceEntryLike,
  AudienceEntryFields,
} from './types/AudienceEntry';
export type {
  BaselineWithExperiencesEntry,
  BaselineWithExperiencesEntryLike,
} from './types/BaselineWithExperiencesEntry';
export type { Entry, EntryLike } from './types/Entry';
export type { EntryFields } from './types/EntryFields';
export type {
  ExperienceEntry,
  ExperienceEntryLike,
  ExperienceFields,
} from './types/ExperienceEntry';
export type {
  ExperimentEntry,
  ExperimentEntryLike,
} from './types/ExperimentEntry';
