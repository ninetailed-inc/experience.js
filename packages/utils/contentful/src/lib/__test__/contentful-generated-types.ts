import { Entry } from 'contentful';

export interface INtAudienceFields {
  /** Name */
  nt_name: string;

  /** Description */
  nt_description?: string | undefined;

  /** Rules */
  nt_rules: Record<string, any>;

  /** Audience Id */
  nt_audience_id: string;
}

/** Ninetailed Audience */

export interface INtAudience extends Entry<INtAudienceFields> {
  sys: {
    id: string;
    type: string;
    createdAt: string;
    updatedAt: string;
    locale: string;
    contentType: {
      sys: {
        id: 'nt_audience';
        linkType: 'ContentType';
        type: 'Link';
      };
    };
  };
}

export interface INtExperienceFields {
  /** Name */
  nt_name: string;

  /** Description */
  nt_description?: string | undefined;

  /** Type */
  nt_type: 'nt_experiment' | 'nt_personalization';

  /** Config */
  nt_config: Record<string, any>;

  /** Audience */
  nt_audience?: INtAudience | undefined;

  /** Variants */
  nt_variants?: Entry<{ [fieldId: string]: unknown }>[] | undefined;
}

/** Ninetailed Experience */

export interface INtExperience extends Entry<INtExperienceFields> {
  sys: {
    id: string;
    type: string;
    createdAt: string;
    updatedAt: string;
    locale: string;
    contentType: {
      sys: {
        id: 'nt_experience';
        linkType: 'ContentType';
        type: 'Link';
      };
    };
  };
}

export interface IHeroFields {
  /** Name */
  name: string;

  /** Ninetailed */
  nt_experiences?: INtExperience[] | undefined;
}

export interface IHero extends Entry<IHeroFields> {
  sys: {
    id: string;
    type: string;
    createdAt: string;
    updatedAt: string;
    locale: string;
    contentType: {
      sys: {
        id: 'hero';
        linkType: 'ContentType';
        type: 'Link';
      };
    };
  };
}
