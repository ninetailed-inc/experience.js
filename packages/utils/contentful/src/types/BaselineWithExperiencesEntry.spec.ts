import { ctfHero } from '../lib/__test__/ctfHero';
import {
  BaselineWithExperiencesEntryLike,
  BaselineWithExperiencesEntrySchema,
} from './BaselineWithExperiencesEntry';
import { IHero, IHeroFields } from './fixtures/contentful';

describe('BaselineWithExperiencesEntry', () => {
  it('should accept the correct types', () => {
    // casting this on purpose to test if the types match
    const data = {} as IHero;

    const fnThatWantsTheCorrectType = (
      baseline: BaselineWithExperiencesEntryLike<IHeroFields>
    ) => {
      return baseline;
    };

    const result = fnThatWantsTheCorrectType(data);

    expect(result).toEqual(data);
  });

  it('should sanitize the entries correctly', () => {
    const result = BaselineWithExperiencesEntrySchema.safeParse(ctfHero);

    expect(result.success).toBe(true);

    if (result.success) {
      expect(result.data.fields.nt_experiences).toHaveLength(1);
      expect(
        result.data.fields.nt_experiences[0].fields.nt_variants
      ).toHaveLength(1);

      expect(
        result.data.fields.nt_experiences[0].fields.nt_variants
      ).toMatchSnapshot();
    }
  });
});
