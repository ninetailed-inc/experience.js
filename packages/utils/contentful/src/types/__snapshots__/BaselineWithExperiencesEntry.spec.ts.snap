// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaselineWithExperiencesEntry should sanitize the entries correctly 1`] = `
Array [
  Object {
    "fields": Object {
      "ctaLink": Array [
        Object {
          "id": "05d26971-eebf-4b89-b459-500e01dc1d0e",
          "label": "Get on the list",
          "type": "internal-url",
          "url": "/products",
        },
      ],
      "description": Object {
        "content": Array [
          Object {
            "content": Array [
              Object {
                "data": Object {},
                "marks": Array [
                  Object {
                    "type": "bold",
                  },
                ],
                "nodeType": "text",
                "value": "Introducing Heinz vintage Drip",
              },
            ],
            "data": Object {},
            "nodeType": "paragraph",
          },
          Object {
            "content": Array [
              Object {
                "data": Object {},
                "marks": Array [],
                "nodeType": "text",
                "value": "It’s NYFW and we’ve teamed up with taste-maker @emmachamberlain to drop a limited run of 157 ketchup-stained, vintage swag pieces. Time to add some flavor to your fashion.",
              },
            ],
            "data": Object {},
            "nodeType": "paragraph",
          },
        ],
        "data": Object {},
        "nodeType": "document",
      },
      "desktopImage": Array [
        Object {
          "bytes": 911720,
          "context": Object {
            "custom": Object {
              "alt": "lady gazing to the right",
            },
          },
          "created_at": "2022-08-16T06:59:26Z",
          "duration": null,
          "format": "png",
          "height": 826,
          "metadata": Array [],
          "original_secure_url": "https://res.cloudinary.com/kraft-heinz-whats-cooking-ca/image/upload/v1660633166/brand-experience/test-images/creative-content-panel/creative-content-panel-hero-banner-desktop_wvdfoa.png",
          "original_url": "http://res.cloudinary.com/kraft-heinz-whats-cooking-ca/image/upload/v1660633166/brand-experience/test-images/creative-content-panel/creative-content-panel-hero-banner-desktop_wvdfoa.png",
          "public_id": "brand-experience/test-images/creative-content-panel/creative-content-panel-hero-banner-desktop_wvdfoa",
          "raw_transformation": "f_auto/q_auto",
          "resource_type": "image",
          "secure_url": "https://res.cloudinary.com/kraft-heinz-whats-cooking-ca/image/upload/f_auto/q_auto/v1660633166/brand-experience/test-images/creative-content-panel/creative-content-panel-hero-banner-desktop_wvdfoa.png",
          "tags": Array [],
          "type": "upload",
          "url": "http://res.cloudinary.com/kraft-heinz-whats-cooking-ca/image/upload/f_auto/q_auto/v1660633166/brand-experience/test-images/creative-content-panel/creative-content-panel-hero-banner-desktop_wvdfoa.png",
          "version": 1660633166,
          "width": 1920,
        },
      ],
      "entryTitle": "[Ninetailed Variant] Home Page Hero Banner",
      "headline": "This is Ninetailed.
<i>wow</i>
",
      "headlineStyle": "headline-4",
      "mobileImage": Array [
        Object {
          "bytes": 647055,
          "context": Object {
            "custom": Object {
              "alt": "lady gazing to the right",
            },
          },
          "created_at": "2022-08-16T06:59:26Z",
          "duration": null,
          "format": "png",
          "height": 820,
          "metadata": Array [],
          "original_secure_url": "https://res.cloudinary.com/kraft-heinz-whats-cooking-ca/image/upload/v1660633166/brand-experience/test-images/creative-content-panel/creative-content-panel-hero-banner-mobile_they3j.png",
          "original_url": "http://res.cloudinary.com/kraft-heinz-whats-cooking-ca/image/upload/v1660633166/brand-experience/test-images/creative-content-panel/creative-content-panel-hero-banner-mobile_they3j.png",
          "public_id": "brand-experience/test-images/creative-content-panel/creative-content-panel-hero-banner-mobile_they3j",
          "raw_transformation": "f_auto/q_auto",
          "resource_type": "image",
          "secure_url": "https://res.cloudinary.com/kraft-heinz-whats-cooking-ca/image/upload/f_auto/q_auto/v1660633166/brand-experience/test-images/creative-content-panel/creative-content-panel-hero-banner-mobile_they3j.png",
          "tags": Array [],
          "type": "upload",
          "url": "http://res.cloudinary.com/kraft-heinz-whats-cooking-ca/image/upload/f_auto/q_auto/v1660633166/brand-experience/test-images/creative-content-panel/creative-content-panel-hero-banner-mobile_they3j.png",
          "version": 1660633166,
          "width": 768,
        },
      ],
      "theme": "red",
      "variant": "content-left",
    },
    "metadata": Object {
      "tags": Array [],
    },
    "sys": Object {
      "contentType": Object {
        "sys": Object {
          "id": "or-creative-content-panel",
          "linkType": "ContentType",
          "type": "Link",
        },
      },
      "createdAt": "2022-09-30T12:41:10.265Z",
      "environment": Object {
        "sys": Object {
          "id": "dev",
          "linkType": "Environment",
          "type": "Link",
        },
      },
      "id": "6Vlth4dgPIflvvWbGo8yd",
      "locale": "en-US",
      "revision": 1,
      "space": Object {
        "sys": Object {
          "id": "ly5kzv4xdfjq",
          "linkType": "Space",
          "type": "Link",
        },
      },
      "type": "Entry",
      "updatedAt": "2022-09-30T12:41:10.265Z",
    },
  },
]
`;
